Source/WebCore/dom/Element.cpp

```diff
@@ -2498,6 +2501,47 @@ void Element::didMoveToNewDocument(Document& oldDocument, Document& newDocument)
             }
         }
     }
+
+    if (langAttrKnownToMatchDocumentElement())
+        oldDocument.removeElementWithLangAttrMatchingDocumentElement(*this);
+
+    updateEffectiveLangState();
+}
+
+void Element::updateEffectiveLangStateFromParent()
+{
+    ASSERT(!hasLanguageAttribute());
+
+    if (auto* parent = parentOrShadowHostElement(); parent && parent != document().documentElement()) {
+        setEffectiveLangKnownToMatchDocumentElement(parent->effectiveLangKnownToMatchDocumentElement());
+        if (UNLIKELY(parent->hasRareData()) && !parent->elementRareData()->effectiveLang().isNull())
+            ensureElementRareData().setEffectiveLang(parent->elementRareData()->effectiveLang());
+        else if (hasRareData())
+            elementRareData()->setEffectiveLang(nullAtom());
+    }
```

Comment:
What happens when the parent element is document element here?

Reply:
We need to handle a transition between "has non-document-element-matching lang" to "no lang" for such elements. I'll add a new subtest.